Add skip links guide (#316)
springernature/frontend-playbook: accessibility/accessibility-checklist.md

* [Tool: HTML5 accessibility feature support for sections](http://www.html5accessibility.com/#sections)

> ### Provide a Skip Navigation option

Reviewer: ```suggestion
### Provide skip links to bypass repetitive content
```

Suggesting this instead of "skip navigation", because there are other things you might want to add a bypass to (e.g. very long, tiresome widgets with 100s of tab stops)

> Not all users are able to navigate by landmarks. For those users the multiple links typically found at the head of a document all have to be read via assistive technology, or laboriously tabbed through using a keyboard.

Reviewer: ```suggestion
Not all users are able to easily navigate by landmarks. Multiple navigation links typically found at the top of a document all have to be laboriously tabbed through using a keyboard, on every page the user visits.
```

Removed the bit about assistive tech just coz everything needs to be read by AT anyway. That changes the focus to how much tedious work it is to physically tab through everything. Also added a bit about how it needs to be done over and over on every page to make it even more 😩

> To help them out you can add a "Skip Navigation" link to the top of each page which, when clicked, will take them to the main section of the page.

Reviewer: ```suggestion
To help them out you can add a "Skip Navigation" link to the top of each page which, when activated, will take them to the main section of the page.
```

We do this:

```
<body>
<p><a href="#maincontent">Skip to main content</a></p>
...

> <main id="maincontent">

Reviewer: One thing that Link do is to add a `tabindex="-1"` to the `main`, which allows it to more consistently receive focus when using the skip link. I feel like we probably need to have that in the example (as not having it will cause bugs), but I'm wary of having both too many explanations, or of having much stuff in the example that isn't explained 🤔 

I wonder if skip links would be a good candidate for a component? Then we could link off to the component at a later date instead of explaining all the bits

Author: Good idea. Added a wee bit about that.

Author: And yeah, I think this is an excellent candidate for a component!

<h1>Heading</h1>
<p>This is the first paragraph</p>
```

> You should supply CSS that hides the Skip Navigation _for sighted users only_, and makes it highly visible when it receives focus via tabbing.

Reviewer: Might be worth adding a bit here about how you can also use skip links to bypass other annoying things here (there's a bit in the WebAIM article about that, under the "multiple skip links" section)

Author: I think that potentially confuses things in what is a very brief section. That WebAIM section itself becomes incredibly nuanced, as they suggest _not_ adding more skip links, and instead using better document structure and in page navigation.

Resources:

> * [https://webaim.org/techniques/skipnav/](WebAIM's overview of Skip Navigation techniques)

Reviewer: ```suggestion
* [Article: WebAIM's overview of Skip Navigation techniques](https://webaim.org/techniques/skipnav/)
```

Author: pls endorse me on linkedin for markdown

> * [https://link.springer.com](Springer Link: an example of a well-done skip navigation)

Reviewer: ```suggestion
* [Example: Springer Link's well-done skip navigation](https://link.springer.com)
```

* [Tool: HTML5 accessibility feature support for sections](http://www.html5accessibility.com/#sections)

### Provide skip links to bypass repetitive content

Not all users are able to easily navigate by landmarks. Multiple navigation links typically found at the top of a document all have to be laboriously tabbed through using a keyboard, on every page the user visits.

To help them out you can add a "Skip Navigation" link to the top of each page which, when activated, will take them to the main section of the page.

We do this:

```
<body>
<p><a href="#maincontent">Skip to main content</a></p>
...
<main id="maincontent" tabindex="-1">
<h1>Heading</h1>
<p>This is the first paragraph</p>
```

> We add `tabindex="-1"` to the targeted element so that it can be visibly focussed when skipped to.

Reviewer: ```suggestion
We add `tabindex="-1"` to the targeted element so that it can be programmatically focussed when skipped to.
```
😉
